include/votca/tools/reducededge.h

```diff
@@ -0,0 +1,94 @@
+/*
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/tools/edge.h>
+
+#ifndef _VOTCA_TOOLS_REDUCEDEDGE_H
+#define _VOTCA_TOOLS_REDUCEDEDGE_H
+
+namespace votca {
+namespace tools {
+
+/**
+ * \brief Connects to vertices
+ *
+ * The edge class stores the ids of two seperate vertices indictating that they
+ * are connected (id1,id2). Unlike a pair the vertex with the lower value is
+ * always placed in id1, this allows us to reduce ambiguity when dealing with
+ * a link.
+ */
+class ReducedEdge : public Edge{
+
+ public:
+  ReducedEdge() {}
+  ~ReducedEdge() {}
+  /// Creates an edge the smallest integer value will be placed in the id1
+  /// spot and the larger in the id2 spot
+  ReducedEdge(std::vector<int> chain);
+  ReducedEdge(int vert1, int vert2) : ReducedEdge(std::vector<int>{vert1,vert2}) {};
+
+  std::vector<int> getChain() const { return vertices_;}
```

Comment:
Are you referring to the getChain() method?

Reply:
yes

Reply:
I think a copy in this case is safer because it maintains the encapsulation. If it was a larger data structure I would use references but for a vector of ints the small performance hit I think is worth it.